pyvista/core/utilities/geometric_objects.py

```diff
@@ -1389,16 +1396,52 @@ def Disc(center=(0.0, 0.0, 0.0), inner=0.25, outer=0.5, normal=(0.0, 0.0, 1.0),
     return surf
 
 
-def Text3D(string, depth=0.5):
+def Text3D(string, width=None, height=None, depth=None, center=(0, 0, 0), normal=(0, 0, 1)):
     """Create 3D text from a string.
 
+    The text may be configured to have a specified width, height or depth.
+
     Parameters
     ----------
     string : str
         String to generate 3D text from.
 
-    depth : float, default: 0.5
-        Depth of the text.
+    width : float, optional
+        Width of the text. If ``None``, the width is scaled
+        proportional to :attr:`height`.
+
+        .. versionadded:: 0.43
+
+    height : float, optional
+        Height of the text. If ``None``, the height is scaled
+        proportional to :attr:`width`.
+
+        .. versionadded:: 0.43
+
+    depth : float, optional
+        Depth of the text. If ``None``, the depth is set to half
+        the :attr:`height` by default. Set to ``0.0`` for planar
+        text.
+
+        .. versionchanged:: 0.43
+
+            The default depth is now calculated dynamcically as
+            half the height. Previously, the default depth had
+            a fixed value of ``0.5``.
```

Comment:
To ensure backward compatibility, could you add the new argument after the old argument `depth`? Specifically, we need to prevent users from assigning the value of `depth` to `width`.

Reply:
Good point. This is addressed in [200235d](https://github.com/pyvista/pyvista/pull/5218/commits/200235dd0fa4d21578b2acf45fe71e4d97bbefdc)